src/mlpack/tests/hpt_test.cpp

```diff
@@ -320,8 +329,8 @@ BOOST_AUTO_TEST_CASE(HPTGradientDescentTest)
   // We pass LARS just because some ML algorithm should be passed. We pass MSE
   // to tell HyperParameterTuner that the objective function (QuadraticFunction)
   // should be minimized.
-  HyperParameterTuner<LARS, MSE, QuadraticFunction, GradientDescent>
-      hpt(a, b, c, d, xMin, yMin, zMin);
+  HyperParameterTuner<LARS, MSE, QuadraticFunction,
+      GradientDescent> hpt(a, b, c, d, xMin, yMin, zMin);
```

Comment:
It looks like the wrong `categoricalDimensions` values end up getting passed to `ens::GradientDescent` here, but I haven't figured out why.  I'll try to take a look into it later tonight.

Reply:
Actually, this is fixed by the `!=` -> `==` changes, but there is one more issue: `CVFunction` doesn't properly unmap values using the `DatasetInfo` object.  This used to be done inside of `GridSearch`, but now that `GridSearch` is a separate part of ensmallen that doesn't know anything about `DatasetInfo`, we have to put that logic inside `CVFunction`.  I'll write up a patch tomorrow.  (Probably while waiting in the registration line for NeurIPS...)

Reply:
Well the registration line was short so I didn't get it done there but here is a patch:

https://gist.github.com/rcurtin/81294d3266175005cd59f4ecdb06e0b7

Reply:
hm, I guess the test passed on your system, so maybe I missed something.

Reply:
Ah, I did not include the `!=` -> `==` changes in my patch---it looks like that is the issue.

Reply:
Yeah, let me fix that.